src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_kds.c

```diff
@@ -200,24 +201,36 @@ static int xocl_add_context(struct xocl_dev *xdev, struct kds_client *client,
 		}
 
 		ret = xocl_icap_lock_bitstream(xdev, &args->xclbin_id);
-		if (ret)
+		if (ret) {
 			goto out;
+		}
 
+		client->ctx->bitstream_locked = true;
```

Comment:
bitstream_locked is only used here in this function, thus, it should not be in a global data structure.